### DISCUSSION TITLE
Own Client Library for documentation

### DISCUSSION THREAD

**Discussion by devskar** [scalar/scalar#3857] [Ideas]
For my Asp.Net core api i have a library that helps to interact with the api. Is it possible to add the client from my library to the scalar documentation of the api?

**Comment by hanspagel** [comment]
Yess! Try x-codeSamples

https://github.com/scalar/scalar/blob/main/documentation/openapi.md#x-codesamples

**Reply by devskar** [reply]
Very awesome! I am sadly failing to find a way to add the x-codeSamples in my .NET Controller. Could you point me in the right direction?

**Reply by xC0dex** [reply]
Hey @devskar, which OpenAPI document generator are you using? Depending on the generator, you have to implement an operation-filter, processor, or transformer.

You could either check for the specific endpoint in the transformer or use an attribute at the controller and fetch this attribute in the transformer.
